src/main/java/net/cardosi/mojo/WatchMojo.java

```diff
@@ -204,16 +207,25 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
         diskCache.release();
 
-        for (CachedProject app : projects.values()) {
-            //TODO instead of N threads per project, combine threads?
+        for (Future f : futures) {
             try {
-                app.watch();
-            } catch (IOException ex) {
-                ex.printStackTrace();
-                //TODO fall back to polling or another strategy
+                f.get();
+            } catch (ExecutionException | InterruptedException e) {
+                e.printStackTrace();
             }
         }
 
+        try {
+            // sleep 500ms, so the printouts aren't muddled
+            Thread.sleep(500);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        FileService watchManager = new FileService(diskCache, projects.values().toArray(new CachedProject[projects.size()]));
```

Comment:
If you do not set the correct size, it will ignore the instance you pass and use reflection to make a new array Element. 
` int size = size();
  T[] r = a.length >= size ? a :
           (T[])java.lang.reflect.Array
           .newInstance(a.getClass().getComponentType(), size);`

I could use toArray() with no arguments, but then what it returns needs casting. So over all I feel this is reasonable.